packages/ui/src/components/organisms/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,173 @@
+  stripe = true,
+  height = 30,
+  stripeDuration = 1000,
+  stripeWidth = 40,
+  style,
+  styleTxt,
+}: ProgressBarProps) => {
+  const Colors = useThemeColor();
+  const percent = total !== 0 ? (value / total) * 100 : 0;
+  const animatedStripe = useRef(new Animated.Value(0)).current;
+  const animatedWidth = useRef(new Animated.Value(0)).current;
+
+  if (Object.keys(colorRepartition).length === 0) {
+    colorRepartition = {
+      0: Colors.errorColor,
+      25: Colors.cautionColor,
+      50: Colors.progressColor,
+      75: Colors.priorityColor,
+      100: Colors.successColor,
+    };
+  }
+
+  let color: Color = Colors.cautionColor;
+  Object.keys(colorRepartition).forEach(key => {
+    const threshold = parseInt(key, 10);
+    if (percent >= threshold && colorRepartition[threshold]) {
+      color = colorRepartition[threshold];
+    }
+  });
+
+  const styles = useMemo(
+    () => getStyles(Colors, height, color, percent, stripeWidth),
+    [Colors, height, color, percent, stripeWidth],
+  );
+
+  useEffect(() => {
+    Animated.timing(animatedWidth, {
+      toValue: percent,
+      duration: 500,
+      useNativeDriver: false,
+    }).start();
+  }, [percent, animatedWidth]);
+
+  useEffect(() => {
+    const stripeAnimation = Animated.loop(
+      Animated.timing(animatedStripe, {
+        toValue: 1,
+        duration: stripeDuration,
+        useNativeDriver: true,
+      }),
+    );
+    stripeAnimation.start();
+  }, [animatedStripe, stripeDuration]);
+
+  const translateX = animatedStripe.interpolate({
+    inputRange: [0, 1],
+    outputRange: [-stripeWidth, Dimensions.get('window').width],
+  });
+
+  return (
+    <View style={[styles.container, style]}>
+      <Animated.View
+        style={[
+          styles.progressBar,
+          {
+            width: animatedWidth.interpolate({
+              inputRange: [0, 100],
+              outputRange: ['0%', '100%'],
+            }),
+          },
+        ]}>
+        {stripe && percent < 100 && (
+          <Animated.View
+            style={[
+              styles.stripe,
+              {
+                transform: [{translateX}],
+              },
+            ]}
+          />
+        )}
+        {showPercent && (
+          <Text style={[styles.text, styleTxt]}>{`${Math.round(
+            percent,
+          )}%`}</Text>
+        )}
+      </Animated.View>
+    </View>
+  );
+};
+
+const getStyles = (
+  Colors: ThemeColors,
+  height: number,
+  color,
+  percent,
+  stripeWidth,
+) =>
+  StyleSheet.create({
+    container: {
+      width: '100%',
+      height: height,
+      borderRadius: 7,
+      elevation: 3,
+      shadowOpacity: 0.5,
+      shadowColor: Colors.secondaryColor.background,
+      shadowOffset: {width: 0, height: 0},
+      backgroundColor: Colors.backgroundColor,
+    },
+    progressBar: {
+      borderRadius: 7,
+      justifyContent: 'center',
+      alignItems: 'center',
+      backgroundColor: color.background,
+      width: `${percent}%`,
```

Comment:
width should not be over 100% even if percent is over 100 otherwise text is getting out of the screen
Use min value between percent and 100 here